schemaregistry/schemaregistry.md

```diff
@@ -0,0 +1,318 @@
+mechanism that allows an implementation to reject updates that do not comply
+with a compatibility policy, if one has been implemented.
+
+### 2.2.1. Multi-format schema groups
+
+If the schema format has not been restricted at the schema group level, each
+schema MAY have its own format. This choice is mutually exclusive. If a format
+has been defined for the group, schemas in the group MUST use that format. 
+
+### 2.2.2. Schema attributes
+
+As per the definition above, the schema object is a collection of versions
+of schema documents. An implementation MAY add further attributes.
+
+#### id
+
+- Type: `String`
+- Description: Identifies the schema.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax
+  - MUST be unique within the scope of the schema registry
+- Examples:
+  - myschema
+  - my-schema
+
+#### format
+
+- Type: `String`
+- Description: Defines the format of this schema. The formats supported by an
+  implementation are specific to the implementation. If the format has been omitted
+  for the schema group, the format MUST be specified at the schema-level. This
+  specification does not mandate support for particular formats.
+- Constraints:
+  - REQUIRED if not set at the schema group level.
+  - MUST be null (not present) if set at the schema group level.
+  - MUST be a non-empty string, if present.
+- Examples:
+  - "avro"
+  - "json-schema"
+  - "xsd11"
+
+#### description
+
+- Type: `String`
+- Description: Explains the purpose of the schema.
+- Constraints:
+  - OPTIONAL
+- Examples:
+  - "This schema describes the toppings for a Pizza."
+
+  #### createdtimeutc
+
+- Type: `Timestamp`
+- Description: Instant when the schema was added to the registry.
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+#### updatedtimeutc
+
+- Type: `Timestamp`
+- Description: Instant when the schema was last updated
+- Constraints:
+  - OPTIONAL
+  - Assigned by the server.
+
+### 2.2.3 Schema version attributes
+
+A schema version is a document. The "body" of a schema version MAY be a text
+document or binary stream. An implementation SHOULD validate whether a
+schema version is valid according to the rules of its format, for instance
+whether it is a valid Avro schema document when the format is Apache Avro.
+
+Within the scope of the schema set, the version is identified by the combination
+of a version number and an optional format identifier. The schema version MAY
+also have an additional, optional unique identifier within the scope of the
+registry.
+
+#### version
+
+- Type: `Integer`
+- Description: The version of the schema. This is a simple counter and tracks
+  the version in the scope of this schema within the schema group. The schema
+  document content MAY indicate a different versioning scheme.
+- Constraints:
+  - REQUIRED
+  - Assigned by server.
+- Examples:
+  - 1
+  - 2
+
+#### id
```

Comment:
How do you see this being used by a consumer of the registry?

Reply:
aka.ms/s/{id}   URL shortener option for greedy protocols.

Reply:
Why isn't that just part of the URL shortener function's logic? E.g. tinyurl.com doesn't ask for an id - I just give it a full URL